lms/lib/comment_client/utils.py

```diff
@@ -171,3 +172,27 @@ def __init__(self, collection, page, num_pages, thread_count=0, corrected_text=N
         self.num_pages = num_pages
         self.thread_count = thread_count
         self.corrected_text = corrected_text
+
+
+def check_forum_heartbeat():
+    """
+    Check the forum connection via its built-in heartbeat service and create an answer which can be used in the LMS
+    heartbeat django application.
+    This function can be connected to the LMS heartbeat checker through the HEARTBEAT_CHECKS variable.
+    """
+    # To avoid dependency conflict
+    from django_comment_common.models import ForumsConfig
+    config = ForumsConfig.current()
+
+    if not config.enabled:
+        # If this check is enabled but forums disabled, don't connect, just report no error
+        return 'forum', True, 'OK'
+
+    try:
+        res = requests.get('%s/heartbeat' % COMMENTS_SERVICE).json()
```

Comment:
Making a `get` request with no timeout could possibly make the request hang indefinitely. I suggest adding a timeout here. If the timeout occurs, the code can catch the exception and report an equivalent heartbeat status. Perhaps the existing `except` will work fine.

Reply:
I added the timeout. Now in case the timeout expires, a connection error is reported:
```
  "forum": {
    "status": false, 
    "message": "HTTPConnectionPool(host='edx.devstack.forum', port=4567): Read timed out. (read timeout=0.01)"
  }, 
```